pallets/crowdloan/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub admin_account: AccountIdOf<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				admin_account: AccountIdOf::<T>::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			AdminAccount::<T>::set(self.admin_account.clone());
+		}
+	}
+
+	/// The administrator account allowed to manage the pallet storage.
+	#[pallet::storage]
+	#[pallet::getter(fn admin_account)]
+	pub type AdminAccount<T> = StorageValue<_, AccountIdOf<T>, ValueQuery>;
+
+	/// The set of contributions.
+	///
+	/// It maps from contributor's account to amount contributed.
+	#[pallet::storage]
+	#[pallet::getter(fn contributions)]
+	pub type Contributions<T> = StorageMap<_, Blake2_128Concat, AccountIdOf<T>, BalanceOf<T>>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// A new admin has been set.
+		/// \[old administrator account, new administrator account\]
+		NewAdminAccountSet(AccountIdOf<T>, AccountIdOf<T>),
+		/// A new contribution has been set.
+		/// \[contributor account, old amount (OPTIONAL), new amount\]
+		NewContributionSet(AccountIdOf<T>, Option<BalanceOf<T>>, BalanceOf<T>),
+		/// A contribution has been removed.
+		/// \[contributor account\]
+		ContributionRemoved(AccountIdOf<T>),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The to-delete contribution is not present.
+		ContributorNotPresent,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Sets a new account as the admin of this pallet.
+		///
+		/// The dispatch origin can be either Sudo or the current admin account.
+		///
+		/// # <weight>
+		/// Weight: O(1)
+		/// - Reads: [Origin Account], AdminAccount
+		/// - Writes: AdminAccount
+		#[pallet::weight(1)]
+		pub fn set_admin_account(
+			origin: OriginFor<T>,
+			new_account: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let who =
+				EnsureOneOf::<AccountIdOf<T>, EnsureRoot<AccountIdOf<T>>, EnsureSigned<AccountIdOf<T>>>::ensure_origin(
+					origin,
+				)?;
+
+			let old_account = AdminAccount::<T>::get();
+
+			if let Either::Right(signed_origin) = who {
+				ensure!(signed_origin == old_account, BadOrigin);
+			}
+
+			let looked_up_account = <T as frame_system::Config>::Lookup::lookup(new_account)?;
+			AdminAccount::<T>::set(looked_up_account.clone());
+
+			Self::deposit_event(Event::NewAdminAccountSet(old_account, looked_up_account));
+
+			Ok(())
+		}
+
+		/// Sets a new contribution amount for a given contributor's account.
+		///
+		/// If a previous contribution is present, it is overridden.
+		///
+		/// The dispatch origin must be the current admin account.
+		///
```

Comment:
M: Missing the emitted event.

```suggestion
		///
		/// Emits `NewContributionSet`.
		///
```